plugin/action/throttle/throttle.go

```diff
@@ -104,18 +153,60 @@ func factory() (pipeline.AnyPlugin, pipeline.AnyConfig) {
 
 func (p *Plugin) Start(config pipeline.AnyConfig, params *pipeline.ActionPluginParams) {
 	p.config = config.(*Config)
+	p.logger = params.Logger
 	p.pipeline = params.PipelineName
-	p.limiterBuff = make([]byte, 0)
+	p.limiterBuf = make([]byte, 0)
 
 	limitersMu.Lock()
-	limiters[p.pipeline] = map[string]*limiter{}
+	limiters[p.pipeline] = map[string]limiter{}
 	limitersMu.Unlock()
 
-	for _, r := range p.config.Rules {
-		p.rules = append(p.rules, NewRule(r.Conditions, complexLimit{r.Limit, r.LimitKind}))
+	format, err := pipeline.ParseFormatName(p.config.TimeFieldFormat)
+	if err != nil {
+		format = p.format
+	}
+	p.format = format
+
+	if p.config.LimiterKind == redisLimiterKind {
+		p.redisClient = redis.NewClient(&redis.Options{
+			Network:  "tcp",
+			Addr:     p.config.RedisHost,
+			Password: p.config.RedisPassword,
+		})
+
+		pingResp := p.redisClient.Ping()
+		if pingResp.Err() != nil {
+			p.logger.Fatalf("can't ping redis: %s", pingResp.Err())
+		}
+	}
+
+	for i, r := range p.config.Rules {
+		p.rules = append(p.rules, NewRule(r.Conditions, complexLimit{r.Limit, r.LimitKind}, i))
+	}
+
+	p.rules = append(p.rules, NewRule(map[string]string{}, complexLimit{p.config.DefaultLimit, p.config.LimitKind}, len(p.config.Rules)))
+}
+
+func (p *Plugin) getNewLimiter(throttleField, limiterKey string, rule *rule) limiter {
+	switch p.config.LimiterKind {
+	case redisLimiterKind:
+		return NewRedisLimiter(
+			p.redisClient,
+			throttleField,
+			limiterKey,
+			p.config.RedisRefreshInterval_,
+			p.config.BucketInterval_,
+			p.config.BucketsCount,
+			rule.limit,
+			p.config.IsThrottleDistributed,
+		)
+	case inMemoryLimiterKind:
+		return NewInMemoryLimiter(p.config.BucketInterval_, p.config.BucketsCount, rule.limit)
+	default:
+		p.logger.Panicf("Unknown limiter kind '%s' - expecting 'redis' or 'memory'", p.config.LimiterKind)
```

Comment:
It will be already descriptive error in config parser. First letter should be in the lowercase. Message should be like `unknown limiter kind: %s`.